src/main/java/com/google/maps/GeoApiContext.java

```diff
@@ -138,6 +143,39 @@ public GeoApiContext(RequestHandler requestHandler) {
         config.supportsClientId, query.toString());
   }
 
+  <T, R extends ApiResponse<T>> PendingResult<T> post(ApiConfig config,
+      Class<? extends R> clazz,
+      GeolocationPostPayload payload) {
```

Comment:
Using `GeolocationPostPayload` makes this method highly specific to that API and thus not reusable. Please make this generic.